mermithid/processors/misc/MultiGasComplexLineShape.py

```diff
@@ -0,0 +1,3089 @@
+        elif self.resolution_function == 'composite_gaussian_scaled':
+            self.results = self.fit_data_composite_gaussian_scaled_fixed_scatter_proportion(freq_bins, data_hist_freq)
+        elif self.resolution_function == 'simulated_resolution_scaled':
+            if self.fit_recon_eff == False:
+                self.results = self.fit_data_simulated_resolution_scaled_fixed_scatter_proportion(freq_bins, data_hist_freq)
+            else:
+                self.results = self.fit_data_simulated_resolution_scaled_fit_recon_eff(freq_bins, data_hist_freq)
+
+        return True
+
+
+    # Establishes a standard energy loss array (SELA) from -1000 eV to 1000 eV
+    # with number of points equal to self.num_points_in_std_array. All convolutions
+    # will be carried out on this particular discretization
+    def std_eV_array(self):
+        emin = -1000
+        emax = 1000
+        array = np.linspace(emin,emax,self.num_points_in_std_array)
+        return array
+
+    # A lorentzian line centered at 0 eV, with 2.83 eV width on the SELA
+    def std_lorenztian_17keV(self):
+        x_array = self.std_eV_array()
+        ans = lorentzian(x_array,0,kr_line_width)
+        return ans
+        
+    #A Dirac delta functin
+    def std_dirac(self):
+        x_array = self.std_eV_array()
+        ans = np.zeros(len(x_array))
+        min_x = np.min(np.abs(x_array))
+        ans[np.abs(x_array)==min_x] = 1.
+        logger.warning('Spectrum will be shifted by lineshape by {} eV'.format(min_x))
+        if min_x > 0.1:
+            logger.warning('Lineshape will shift spectrum by > 0.1 eV')
+        if min_x > 1.:
+            logger.warning('Lineshape will shift spectrum by > 1 eV')
+            raise ValueError('problem with std_eV_array()')
+        return ans
+        
+    # A gaussian function
+    def gaussian(self, x_array, A, sigma, mu):
+        f = A*(1./(sigma*np.sqrt(2*np.pi)))*np.exp(-(((x_array-mu)/sigma)**2.)/2.)
+        return f
+
+    # A gaussian centered at 0 eV with variable width, on the SELA
+    def std_gaussian(self, sigma):
+        x_array = self.std_eV_array()
+        ans = ComplexLineShapeUtilities.gaussian(x_array,1,sigma,0)
+        return ans
+
+    def composite_gaussian(self):
+        x_array = self.std_eV_array()
+        ans = 0
+        A_array = self.A_array
+        sigma_array = self.sigma_array
+        for A, sigma in zip(A_array, sigma_array):
+            ans += self.gaussian(x_array, A, sigma, 0)
+        return ans
+
+    def composite_gaussian_pedestal_factor(self, pedestal_factor):
+        x_array = self.std_eV_array()
+        ans = 0
+        A_array = self.A_array
+        sigma_array = np.array(self.sigma_array)
+        sigma_array = sigma_array*[1, 1, pedestal_factor, 1]
+        for A, sigma in zip(A_array, sigma_array):
+            ans += self.gaussian(x_array, A, sigma, 0)
+        return ans
+
+    def composite_gaussian_scaled(self, scale_factor):
```

Comment:
the shapes that we are not using (smeared triangles and maybe others) should maybe not be merged to develop.